### PR TITLE
Api 25622 tracked item status

### PR DESCRIPTION
**Pull request by FonzMP**
## Summary

-  Reworks tracked_items in v2 claims show

## Related issue(s)
- https://vajira.max.gov/browse/API-25622

## Testing done
- RSpec and updated cassette

## What areas of the site does it impact?
- Claims V2

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by stiehlrod — COMMENTED** [review]
Tested with `{{uri}}/services/claims/v2/veterans/1012826664V603033/claims/285456`, and am getting this stack trace:
```                "exception": "no implicit conversion of Symbol into Integer",
                "backtrace": [
                    "/Users/jennicastiehl/Oddball_Github/vets-api/modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:495:in `block in find_tracked_item'",```

**Comment by ryan-mcneil** [comment]
@FonzMP @stiehlrod Please let me know once you've tested again and are happy with it, and I'll approve!

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 1]
```diff
@@ -373,110 +373,130 @@ def waiver_boolean(filed5103_waiver_ind)
...
+          tracked_item[:suspns_dt].present? && tracked_item[:accept_dt].nil?
+        end
+
+        def build_tracked_item(tracked_item, status, item, uploaded)
+          uploads_allowed = uploads_allowed?(status)
+          {
+            closed_date: date_present(tracked_item[:accept_dt]),
+            description: item[:items],
+            display_name: tracked_item[:short_nm],
+            overdue: overdue?(tracked_item), # EVSS generates this field
```
Comment (mchristiansonVA) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:479]: Should we remove this comment given that we're determining this via the overdue?(tracked_item) method?

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 2]
```diff
@@ -373,110 +373,130 @@ def waiver_boolean(filed5103_waiver_ind)
...
+            received_date: date_present(tracked_item[:receive_dt]),
+            requested_date: tracked_item_req_date(tracked_item, item),
+            status:, # EVSS generates this field
+            suspense_date: date_present(tracked_item[:suspns_dt]),
+            id: tracked_item[:dvlpmt_item_id].to_i,
+            uploaded:, # EVSS generates this field
+            uploads_allowed: # EVSS generates this field
```
Comment (mchristiansonVA) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:486]: Same question on this comment, we're determining this via uploads_allowed?(status).

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb (hunk 1 above)
Comment (FonzMP) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb]: Removed all of these comments. 👍

**Comment by FonzMP** [comment]
> Tested with `{{uri}}/services/claims/v2/veterans/1012826664V603033/claims/285456`, and am getting this stack trace: `"exception": "no implicit conversion of Symbol into Integer",` `` "backtrace": [ "/claims_controller.rb:495:in `block in find_tracked_item'", ``

@stiehlrod Returning a hash for tracked_items. I did not test on differing claims before pushing this so nice catch! I'll do a few more locally here, but these should handle the lone tracked item.

**Review by stiehlrod — APPROVED** [review]
Tested: `{{uri}}/services/claims/v2/veterans/1012666073V986297/claims/600320269`, `{{uri}}/services/claims/v2/veterans/1012666073V986297/claims/600334308`, and `{{uri}}/services/claims/v2/veterans/1012666073V986297/claims/600336398`

**Pull request merged**